alerts_consumer.go

```diff
@@ -202,24 +250,37 @@ func (c *AlertsConsumer) SendBatch(batch [][]byte, tag string) error {
 		}
 	}
 
+	var err error
 	retry := retrier.New(retrier.ExponentialBackoff(5, 50*time.Millisecond), nil)
-	retry.Run(func() error {
+
+	err = retry.Run(func() error {
 		lg.TraceD("sfx-add-datapoints", logger.M{"point-count": len(pts)})
 		return c.sfxSink.AddDatapoints(context.Background(), pts)
 	})
-	var err error
 	if err != nil && err.Error() == "invalid status code 400" { // internal buffer full in sfx
 		return kbc.PartialSendBatchError{ErrMessage: "failed to add datapoints: " + err.Error(), FailedMessages: batch}
 	}
 
-	retry = retrier.New(retrier.ExponentialBackoff(5, 50*time.Millisecond), nil)
-	retry.Run(func() error {
+	err = retry.Run(func() error {
 		lg.TraceD("sfx-events", logger.M{"point-count": len(evts)})
 		return c.sfxSink.AddEvents(context.Background(), evts)
 	})
 	if err != nil && err.Error() == "invalid status code 400" { // internal buffer full in sfx
 		return kbc.PartialSendBatchError{ErrMessage: "failed to add events: " + err.Error(), FailedMessages: batch}
 	}
 
+	err = retry.Run(func() error {
```

Comment:
Should we be using a different `retrier` with parameters tuned for the AWS api?

Reply:
Good idea, I'll look at what AWS suggests.

Reply:
FWIW aws-sdk-go has built-in retry behavior, so it might not be necessary to do this

Reply:
zerp - totally forgot about that 👍